roles/provision_virtual_esxi/README.md

```diff
@@ -0,0 +1,92 @@
+# provision_virtual_esxi
+
+Provision one or more virtual ESXi hosts
+
+## Requirements
+
+pyvomi
+
+## Role Variables
+
+### Auth
+- **provision_virtual_esxi_username**:
+  - str, The username to use to authenticate to the esxi or vcenter on which you want to deploy the vm. Required.
+
+- **provision_virtual_esxi_password**:
+  - str, The password to use to authenticate to the esxi or vcenter on which you want to deploy the vm. Required.
+
+- **provision_virtual_esxi_validate_certs**:
+  - bool, If true then certificates will be validated when connecting to the esxi or vcenter for auth. Optional.
+
+### Placement
+- **provision_virtual_esxi_cluster**:
+  - str, The name of the cluster in which you want to deploy the new vms.
+
+- **provision_virtual_esxi_datacenter**:
+  - str, The name of the datacenter in which you want to deploy the new vms.
+
+- **provision_virtual_esxi_resource_pool**:
+  - str, The name of the resource pool in which to place the VMs. Its recommended to use a resource pool and limit the amount of resources these ESXi hosts can use
+
+- **provision_virtual_esxi_folder**:
+  - str, The name of the folder in which to place the VMs
+
+
+### VM Options
+
+- **provision_virtual_esxi_vm_names**:
+  - list(str), A list of names corresponding to esxi hosts you want to manage. If no VM exists with the name, a new VM will be created. VMs are created in booted to save time.
+
+- **provision_virtual_esxi_datastore_iso_path**:
+  - str, The datastore path to the ESXi ISO file that new VMs should use to boot. For example, a file in the folder ISO on datastore1 might have the path `[datastore1] ISO\my_esxi_8.iso`
+
+- **provision_virtual_esxi_networks**:
+  - list(dict), A list of dictionaries describing the network device configs for this VM. Required. You may need to enable promiscuous mode and allow forged transmits on the upstream port group/vSwitch to have a working network downstream. See https://docs.ansible.com/ansible/latest/collections/community/vmware/vmware_guest_module.html#parameter-networks
+
+- **provision_virtual_esxi_disks**:
+  - list(dict), A list of dictionaries describing the disk device configs for this VM. See https://docs.ansible.com/ansible/latest/collections/community/vmware/vmware_guest_module.html#parameter-disk
+
+- **provision_virtual_esxi_memory_mb**:
+  - int, The amount of memory to assign to this VM. For example, 18000
+
+- **provision_virtual_esxi_cpus**:
+  - int, The number of vCPUs to assign to this VM. For example, 4
+
+## Dependencies
+
+- NA
+
+## Example Playbook
+```yaml
+---
+- hosts: localhost
+  roles:
+    - role: cloud.vmware_ops.provision_virtual_esxi
+      vars:
+        provision_virtual_esxi_hostname: "{{ vcenter_hostname }}"
+        provision_virtual_esxi_username: "{{ vcenter_username }}"
+        provision_virtual_esxi_password: "{{ vcenter_password }}"
+        provision_virtual_esxi_folder: ""
+        provision_virtual_esxi_names:
```

Comment:
Did you consider to have list of dict to specify VM specification instead of list of names and share rest of variables?
I can imagine user mostly would use template instead of ISO, so maybe not good case. But maybe there is case where user don't want to share same spec for all VMs?

Reply:
I did think about that; I thought a dictionary might be harder to document. But I could see a user wanting to deploy hosts with different specs so ill implement something like you suggest.

Also I did not think of deploying from a template. Maybe I will change this role name to be specific to ISO for now

Reply:
I changed the variable to `provision_virtual_esxi_vms` and made it a list that accepts different parameters like  memory or cpu. If these values are not set for the VM, the role will default to using the corresponding role parameter like `provision_virtual_esxi_memory_mb` or `provision_virtual_esxi_cpus`